libp2p/stream/lpstream.nim

```diff
@@ -80,6 +80,11 @@ type
     opened*: uint64
     closed*: uint64
 
+proc `$`*(dir: Direction): string =
```

Comment:
Why is this necessary?

Reply:
To do this in the dialer `doAssert not isNil(upgradedConn), "connection died after upgrade " & $upgradeDir`

Reply:
Sounds like a bug / missing import / whatever, `$` should be autogenerated:
```nim
type Xxx {.pure.} = enum
  A, B, C

#works
echo $Xxx.A
```

Reply:
I had an issue with that, but now I can't reproduce it. Removed.